### PR TITLE
Add cell id checking to avoid runtime error

### PR DESCRIPTION
**Pull request by rickiepark**
Surprisingly, some cells doesn't have cell id. :(
So, I just want to check if it's exist and reset cell id to blank.

Thanks.

### CODE REVIEW

**Comment by rickiepark** [comment]
Oops,
Hi @jaketae 
I ask you one-more-consent. 🙏 😃

**Comment by jaketae** [comment]
@googlebot I consent.

File: tools/nb_code_sync.py [hunk 1]
```diff
@@ -123,6 +123,8 @@ def _is_source_code_equal(x_list, y_list):
 
   def _set_cell_source(self, cell_id, source):
     for i, cell in enumerate(self.data["cells"]):
+      if "id" not in cell["metadata"]:
```
Comment (jaketae) [inline tools/nb_code_sync.py:126]: I know this is not a translation, but I wanted to leave a quick suggestion. How about putting everything in a `try` statement?

```
try:
    if cell["metadata"]["id"] == cell_id:
        self.data["cells"][i]["source"] = source
        break
except RuntimeError: # Or any other error that the code block produces
    cell["metadata"]["id"] = ""
```

**Comment by jaketae** [comment]
I know this is not a translation, but I wanted to leave a quick suggestion. How about putting everything in a `try` statement?

```
try:
    if cell["metadata"]["id"] == cell_id:
        self.data["cells"][i]["source"] = source
        break
except RuntimeError: # Or any other error that the code block produces
    cell["metadata"]["id"] = ""
```

And thank you so much for updating the other notebooks!

**Comment by lamberta** [comment]
Ah, yes—we've discussed this before. Cell `id` is a Colab thing and not part of the Jupyter notebook spec, though it definitely makes keeping track of translations easier.

File: tools/nb_code_sync.py [hunk 2]
```diff
@@ -123,11 +123,12 @@ def _is_source_code_equal(x_list, y_list):
 
   def _set_cell_source(self, cell_id, source):
     for i, cell in enumerate(self.data["cells"]):
-      if "id" not in cell["metadata"]:
+      try:
+        if cell["metadata"]["id"] == cell_id:
```
Comment (lamberta) [inline tools/nb_code_sync.py:127]: I think we can revert the last commit and just change this to:
`if cell["metadata"].get("id") == cell_id:`
WIll that work?

Comment (rickiepark) [reply tools/nb_code_sync.py:127]: Sure, it works! :-D 👍

**Pull request merged**